src/GafferOSL/ShadingEngine.cpp

```diff
@@ -299,10 +299,33 @@ class RenderState
 		{
 			m_pointIndex++;
 		}
+	
+		bool getMatrixToObject( OIIO::ustring name, Imath::M44f &result ) const
+		{
+			ShadingEngine::Transforms::const_iterator i = m_transforms.find( name );
+			if( i != m_transforms.end() )
+			{
+				result = i->second.toObjectSpace;
+				return true;
+			}
+			return false;
+		}
+
+		bool getMatrixFromObject( OIIO::ustring name, Imath::M44f &result ) const
+		{
+			ShadingEngine::Transforms::const_iterator i = m_transforms.find( name );
+			if( i != m_transforms.end() )
+			{
+				result = i->second.fromObjectSpace;
+				return true;
+			}
+			return false;
+		}
 
 	private :
 
 		size_t m_pointIndex;
+		const ShadingEngine::Transforms &m_transforms;
```

Comment:
Since we're not going to use `ustring` in the public API, we'll need something else here. I'd suggest `boost::flat_map<ustring, Transform>`, since it seems to have pretty good lookup performance, especially for smallish numbers of keys (we use it in the Context).